Cargo.toml

```diff
@@ -58,6 +58,7 @@ clippy.large_include_file = "warn"
 clippy.large_stack_arrays = "warn"
 clippy.mismatching_type_param_order = "warn"
 clippy.missing_fields_in_debug = "warn"
+clippy.needless_bitwise_bool = "warn"
```

Comment:
```suggestion
```
Given that the lint is functionally useless, I'd just do the targeted fix

Reply:
I mean I wouldn't say useless, it would catch cases such as:
```rust
if true & false {
}
```

But it could of course be better I think.

Reply:
I'm not sure it even will catch that, because those RHSed aren't potentially expensive.

Reply:
Oh right, it doesn't, thought I tested this, it works with (which I think is what I tested)
```rust
if true & !false {}
```

So yeah not really useful in those rare cases where it triggers, the last comment in the linked issue with something like:

```rust
fn foo2(x1: i32, x2: i32) {
    if (x1 < x2) & (x1 != 0) {}
}
```
doesn't even trigger this lint. Alright in that case I'm focusing this to the actual fix...